tenders/hvt/hvt_openbsd.c

```diff
@@ -184,12 +185,35 @@ int hvt_guest_mprotect(void *t_arg, uint64_t addr_start, uint64_t addr_end,
      * Host-side page protections:
      *
      * Ensure that guest-executable pages are not also executable in the host.
-     *
+     */
+    if(mprotect(vaddr_start, size, prot & ~(PROT_EXEC)) == -1)
+        return -1;
+
+    ret = 0;
+#if defined(VMM_IOC_MPROTECT_EPT)
+    /*
      * Guest-side page protections:
-     *
-     * Manipulating guest-side (EPT) mappings is currently not supported by
-     * OpenBSD vmm, so there is nothing more we can do.
      */
-    prot &= ~(PROT_EXEC);
-    return mprotect(vaddr_start, size, prot);
+    struct hvt_b *hvb = hvt->b;
+    struct vm_mprotect_ept_params *vmep;
+
+    vmep = calloc(1, sizeof (struct vm_mprotect_ept_params));
+    if (vmep == NULL) {
+        warn("calloc");
+        return -1;
+    }
+
+    vmep->vmep_vm_id = hvb->vcp_id;
+    vmep->vmep_vcpu_id = hvb->vcpu_id;
+
+    // sgpa = vmr->vmr_gpa(0x0) + (addr - vmr->vmr_va)
```

Comment:
Nit: This looks like a left-over comment from development?

Reply:
@mato this has been removed.